app.py

```diff
@@ -83,7 +90,13 @@ async def run_groups_authn():
         request = Request()
         creds.refresh(request)
 
-    # sync IAM groups to Cloud SQL instances
-    await groups_sync(iam_groups, sql_instances, creds, private_ip)
-
+    try:
+        # sync IAM groups to Cloud SQL instances
+        await groups_sync(iam_groups, sql_instances, creds, group_roles, private_ip)
```

Comment:
nit: consider merging iam_groups and group_roles. you can do the truncation before passing in here

Reply:
I don't think I can truncate at this stage, to get the list of IAM users of an IAM group, I need the full IAM email address later on.